In [1]:
import pandas as pd
import numpy as np

In [2]:
data = pd.read_csv("Desktop/Quiz1.csv")

In [4]:
data

STITCH      Compound                   Unnamed: 2
0              1  CID003062316                    Dasatinib
1              2  CID003062316                    Dasatinib
2              3  CID003062316                    Dasatinib
3              4  CID003062316                    Dasatinib
4              5  CID003062316                    Dasatinib
...          ...           ...                          ...
1048570  1048571  CID000003310  Epipodophyllotoxin VP-16213
1048571  1048572  CID000003310  Epipodophyllotoxin VP-16213
1048572  1048573  CID000003310  Epipodophyllotoxin VP-16213
1048573  1048574  CID000003310  Epipodophyllotoxin VP-16213
1048574  1048575  CID000003310  Epipodophyllotoxin VP-16213

[1048575 rows x 3 columns]

In [7]:
data.columns

Index(['STITCH', 'Compound', 'Unnamed: 2'], dtype='object')

In [8]:
data['STITCH']

0                1
1                2
2                3
3                4
4                5
            ...   
1048570    1048571
1048571    1048572
1048572    1048573
1048573    1048574
1048574    1048575
Name: STITCH, Length: 1048575, dtype: int64

In [9]:
del data['STITCH']

In [10]:
data

Compound                   Unnamed: 2
0        CID003062316                    Dasatinib
1        CID003062316                    Dasatinib
2        CID003062316                    Dasatinib
3        CID003062316                    Dasatinib
4        CID003062316                    Dasatinib
...               ...                          ...
1048570  CID000003310  Epipodophyllotoxin VP-16213
1048571  CID000003310  Epipodophyllotoxin VP-16213
1048572  CID000003310  Epipodophyllotoxin VP-16213
1048573  CID000003310  Epipodophyllotoxin VP-16213
1048574  CID000003310  Epipodophyllotoxin VP-16213

[1048575 rows x 2 columns]

In [11]:
data.rename(columns={'Compound':'STITCH'},inplace=True)
data

STITCH                   Unnamed: 2
0        CID003062316                    Dasatinib
1        CID003062316                    Dasatinib
2        CID003062316                    Dasatinib
3        CID003062316                    Dasatinib
4        CID003062316                    Dasatinib
...               ...                          ...
1048570  CID000003310  Epipodophyllotoxin VP-16213
1048571  CID000003310  Epipodophyllotoxin VP-16213
1048572  CID000003310  Epipodophyllotoxin VP-16213
1048573  CID000003310  Epipodophyllotoxin VP-16213
1048574  CID000003310  Epipodophyllotoxin VP-16213

[1048575 rows x 2 columns]

In [12]:
data.rename(columns={'Unnamed: 2':'Compound'},inplace=True)
data

STITCH                     Compound
0        CID003062316                    Dasatinib
1        CID003062316                    Dasatinib
2        CID003062316                    Dasatinib
3        CID003062316                    Dasatinib
4        CID003062316                    Dasatinib
...               ...                          ...
1048570  CID000003310  Epipodophyllotoxin VP-16213
1048571  CID000003310  Epipodophyllotoxin VP-16213
1048572  CID000003310  Epipodophyllotoxin VP-16213
1048573  CID000003310  Epipodophyllotoxin VP-16213
1048574  CID000003310  Epipodophyllotoxin VP-16213

[1048575 rows x 2 columns]

In [13]:
df = data
df

STITCH                     Compound
0        CID003062316                    Dasatinib
1        CID003062316                    Dasatinib
2        CID003062316                    Dasatinib
3        CID003062316                    Dasatinib
4        CID003062316                    Dasatinib
...               ...                          ...
1048570  CID000003310  Epipodophyllotoxin VP-16213
1048571  CID000003310  Epipodophyllotoxin VP-16213
1048572  CID000003310  Epipodophyllotoxin VP-16213
1048573  CID000003310  Epipodophyllotoxin VP-16213
1048574  CID000003310  Epipodophyllotoxin VP-16213

[1048575 rows x 2 columns]

In [14]:
df['STITCH'].value_counts()

CID000002083    15383
CID000000853    13261
CID000001983    12923
CID000000444    12558
CID000002662    11177
                ...  
CID000005482       19
CID000153941       18
CID000002551       10
CID005282044       10
CID000002673        6
Name: STITCH, Length: 632, dtype: int64

In [15]:
df['Compound'].value_counts()

Salbutamol       15383
DL-Thyroxine     13261
Acetaminophen    12923
Bupropion        12558
Celecoxib        11177
                 ...  
Procainamide        25
Pemoline            23
Tioconazole         19
Carbastat           10
Cefradin             6
Name: Compound, Length: 610, dtype: int64

No. of unique STITCH values = 632 and Compound values = 610

In [16]:
df.groupby('Compound')['STITCH'].nunique()

Compound
((1R)-3-Methyl-1-(((2S)-3-phenyl-2-((pyrazinylcarbonyl)amino)propanoyl)amino)butyl)boronic acid    1
((1S,3S)-3-Isopropylcyclohexane-1-carbonyl)-D-phenylalanine                                        1
(+)-Tamsulosin                                                                                     1
(+/-)-Ambrisentan                                                                                  1
(-)-Quinine                                                                                        1
                                                                                                  ..
delta9-Tetrahydrocannabinol                                                                        1
gamma-Aminobutyric acid                                                                            1
meclizine                                                                                          1
tazarotene                                                                        

In [25]:
ds = df.groupby('Compound')['STITCH']

In [26]:
ds

In [27]:
data

STITCH                     Compound
0        CID003062316                    Dasatinib
1        CID003062316                    Dasatinib
2        CID003062316                    Dasatinib
3        CID003062316                    Dasatinib
4        CID003062316                    Dasatinib
...               ...                          ...
1048570  CID000003310  Epipodophyllotoxin VP-16213
1048571  CID000003310  Epipodophyllotoxin VP-16213
1048572  CID000003310  Epipodophyllotoxin VP-16213
1048573  CID000003310  Epipodophyllotoxin VP-16213
1048574  CID000003310  Epipodophyllotoxin VP-16213

[1048575 rows x 2 columns]

In [28]:
data.to_csv('Desktop/Quiz-1.csv')